pkg/storagesvc/archivePruner.go

```diff
@@ -142,9 +143,11 @@ func (pruner *ArchivePruner) getOrphanArchives(ctx context.Context) {
 // Start starts a go routine that listens to a channel for archive IDs that need to deleted.
 // Also wakes up at regular intervals to make a list of archive IDs that need to be reaped
 // and sends them over to the channel for deletion
-func (pruner *ArchivePruner) Start(ctx context.Context) {
+func (pruner *ArchivePruner) Start(ctx context.Context, mgr manager.Interface) {
 	ticker := time.NewTicker(pruner.pruneInterval * time.Minute)
-	go pruner.pruneArchives()
+	mgr.Add(ctx, func(_ context.Context) {
+		pruner.pruneArchives()
```

Comment:
We need to pass context to the inner function and make sure it exits on context.Done otherwise goroutine won't exit.